dcos/mesos.py

```diff
@@ -987,8 +992,35 @@ def __init__(self, task, cmd, interactive=False, pty=False):
         self.output_queue = Queue()
         self.exit_queue = Queue()
 
-    def _initialize_exec_stream(self):
-        pass
+    def initialize_exec_stream(self):
+        headers = {'connection': 'keep-alive',
+                   'content-type': 'application/x-protobuf'}
+
+        # Call requires an initialization message
+        launch_nested_container_msg = agent_pb2.LaunchNestedContainerSession(
+            self.container_id,
+            self.cmd,
+            self.tty,
+            self.interactive
+        )
+        # Initilize nested container using Call
+        call_msg = agent_pb2.Call(
+            agent_pb2.Call.LAUNCH_NESTED_CONTAINER_SESSSION,
+            launch_nested_container_msg)
+
+        pickled_msg = pickle.dumps(call_msg)
+
+        request = requests.Request(
```

Comment:
we should be using `dcos.http` for requests. Its our wrapper around requests- handles auth, ssl, etc.